RoosterPlanner.Api/Controllers/TasksController.cs

```diff
@@ -150,7 +152,7 @@ public ActionResult UpdateTask(TaskViewModel taskViewModel)
 
                 if (updatedTask.CategoryId != null && oldTask.CategoryId != updatedTask.CategoryId)
                 {
-                    oldTask.Category = taskService.GetCategory(updatedTask.CategoryId??Guid.Empty).Result.Data;
+                    oldTask.Category = taskService.GetCategory(updatedTask.CategoryId ?? Guid.Empty).Result.Data;
```

Comment:
Als de GetCategory een Async methode is (public async Task<> GetCategory), dan na de "oldTask.Category = " een await gebruiken ipv .Result  (dan zou het worden GetAwaiter().GetResult()

Reply:
Ik kan er van maken:
<code>oldTask.Category =  taskService.GetCategory(updatedTask.CategoryId ?? Guid.Empty).GetAwaiter().GetResult().Data;</code>
**of**
<code>oldTask.Category = (await taskService.GetCategory(updatedTask.CategoryId ?? Guid.Empty)).Data;</code>



<code>oldTask.Category = await taskService.GetCategory(updatedTask.CategoryId ?? Guid.Empty).GetAwaiter().GetResult();</code>
**vindt hij niet leuk**

Reply:
Als je await gebruikt, dan geen GetAwaiter().GetResult()

Waarschijnlijk is CategoryId een Guid, dus "updatedTask.CategoryId != null" => "updatedTask.CategoryId != Guid.Empty"